dconf.py

```diff
@@ -12,15 +12,15 @@ def _check_output_strip(command):
 def _escape_single_quotes(string):
     return re.sub("'", r"'\''", string)
 
-def _get_dbus_bus_address(user):
+def _get_dbus_bus_address(user, dm):
     if user is None:
         if environ.get('DBUS_SESSION_BUS_ADDRESS') is None:
             return None
 
         return "DBUS_SESSION_BUS_ADDRESS={}".format(environ['DBUS_SESSION_BUS_ADDRESS'])
 
     try:
-        pid = _check_output_strip(['pgrep', '-u', user, 'gnome-session'])
+        pid = _check_output_strip(['pgrep', '-f', '-u', user, "^" + dm])
```

Comment:
Hi, thanks for the PR, adding a DM choice looks good, however this pattern doesn't work for me on Fedora Silverblue (i just added -a to print the matching commands):

```
$ pgrep -a -f -u jistr '^gnome-session'


$ pgrep -a -u jistr gnome-session
1799 /usr/libexec/gnome-session-binary
1860 /usr/libexec/gnome-session-ctl --monitor
1863 /usr/libexec/gnome-session-binary --systemd-service --session=gnome
```
Does that anchor really work for you? pgrep should match on a full path with -f, no?

What about this:

```
$ pgrep -a -f -u jistr '^[^ ]*/gnome-session'
1799 /usr/libexec/gnome-session-binary
1860 /usr/libexec/gnome-session-ctl --monitor
1863 /usr/libexec/gnome-session-binary --systemd-service --session=gnome
```

So that RE would anchor it to the beginning, then accept anything except space (to make sure we're truly matching the executable and not some parameter later on), then slash to make sure we're matching the beginning of the executable, and then the actual supplied `dm` parameter.

Reply:
> Does that anchor really work for you?

Yes, the anchor does work:

```bash
$ pgrep -f -u rich ^cinnamon-session
10738
$ pgrep -a -f -u rich ^cinnamon-session
10738 cinnamon-session --session cinnamon
```

> pgrep should match on a full path with -f, no?

Originally, I tested as-is (w/o the `-f`), then I googled around and found that it wasn't working because pgrep only matches the [first 15 characters](https://unix.stackexchange.com/a/267021):

```bash
$ pgrep -a -u rich ^cinnamon-session
$ ps -ef | grep cinnamon-session
rich      3408 21542  0 21:59 pts/126  00:00:00 grep --color cinnamon-session
rich     10738 10697  0 Feb13 ?        00:00:08 cinnamon-session --session cinnamon
rich     10804 10738  0 Feb13 ?        00:00:04 /usr/bin/ssh-agent /usr/bin/dbus-launch --exit-with-session /usr/bin/im-launch cinnamon-session-cinnamon
rich     10807     1  0 Feb13 ?        00:00:00 /usr/bin/dbus-launch --exit-with-session /usr/bin/im-launch cinnamon-session-cinnamon
21:59:16 rich@s27lxlap [~]
```

> What about this:

Doesn't appear to work:

```bash
$ pgrep -a -f -u rich '^[^ ]*/cinnamon-session'
```

I think the root cause may be that my DM is resoled via $PATH and the output provided is already qualified as an absolute path.  The original script doesn't have this problem, as pgrep only matches process name, unless --full is provided =/ ...

Maybe this problem can be pushed to the left by modifying the proposed `dm` paramter to be the whole regex pattern?

Reply:
Sorry for the delay, it's hard to find time for me these days. +1 for making the dm parameter a regex. I think a good default would be `^[^ ]*/gnome-session` as shown above, so that we can add the `-f` grep option?